functions/src/postActions.js

```diff
@@ -89,19 +87,19 @@ exports.downVote = functions.https.onCall(async (data, context) => {
 
   // Get current user
   const {_, userDocSnap} = await fetchAndValidateUser(uid);
+
   // Get target post
   const {postRef, postDocSnap} = await fetchAndValidatePost(postId);
 
   // Verify that user has access to post.
   verifyUserMeetsTokenRequirements(userDocSnap.data(), postDocSnap.data()); 
 
-  // Add user to upVote list if user is not already in upvote list.
+  // Add user to up vote list if user is not already in upvote list.
   const downVotes = postDocSnap.data().downVoteUserIds;
   if (downVotes.indexOf(uid) === -1) {
     try {
-      downVotes.push(uid);
       const upVotes = postDocSnap.data().upVoteUserIds.filter((item) => item !== uid);
-      await postRef.set({ upVoteUserIds: upVotes, downVoteUserIds: downVotes }, { merge: true });
+      await postRef.set({ upVoteUserIds: upVotes, downVoteUserIds: [uid, ...downVotes] }, { merge: true });
```

Comment:
https://firebase.google.com/docs/firestore/manage-data/add-data#update_elements_in_an_array
in case but doesn t matter

Reply:
Great recommendation, thanks!